README.md

```diff
@@ -1,164 +1,190 @@
+      new L.latLng(51.52,-0.10),
+      new L.latLng(51.52,-0.14),
+      new L.latLng(51.50,-0.10),
+      new L.latLng(51.50,-0.14)
+    ],
+    fullResolutionSrc: 'large.jpg', // optionally pass in a higher resolution image to use in full-res exporting
+    suppressToolbar: false // defaults to false, but you can turn off the toolbar interface and build your own
+  }
+).addTo(map);
+
+L.DomEvent.on(img._image, 'load', img.editing.enable, img.editing); // enable editing
+
+```
+
+## Full-resolution download
+
+We've added a GPU-accelerated means to generate a full resolution version of the distorted image; it requires two additional dependencies to enable; see how we've included them in the demo:
+
+
+```
+<script src="../node_modules/webgl-distort/dist/webgl-distort.js"></script>
+<script src="../node_modules/glfx-js/dist/glfx.js"></script>
+```
+## "Reveal" functionality
+
+The "reveal" tool extends the native `EditOverlayAction` Leaflet class behaviour and hides some user-specified tools, hence saving up space in the toolbar and giving more room to your mapview! Reference the comments below for a detailed explaination on what happens when the "reveal" tool is executed.
+
+```js
+function() {
+
+  /* specify an array of tools that will initially be
+   hidden inside the reveal tool in the toolbar
+   also, input is taken from the exposed API in the HTML */
+  var hidden_tools = this._overlay.options.hidden_tools, i;
+
+  /* On reveal, add all the "hidden" tools to
+  the toolbar */
+  for (i = 0; i < hidden_tools.length; i++) {
+    this._overlay._addTool(hidden_tools[i]);
+  }
+
+  /* Remove "reveal" tool from the toolbar */
+  this._overlay._removeTool(ToggleReveal);
+
+  this._overlay.editing._hideToolbar();
+}
+```
+## Adding Custom Toolbars
+
+You also have the ability to incorporate additional toolbars into view by defining appropriate fields in the exposed `custom_toolbar` API inside your HTML, for eg., we at Public Lab use a custom keymapping guide toolbar by defining it as,
+```js
+var keymapper = {
+  position: "topright",
+  el: "div",
+  styleClass: "l-container",
+  DOMString:
+    "<b><center><h3>Keymappings</h3><center><hr/></center></center><ul><li>L: Lock overlay</li><li>O: Outline overlay</li><li>R: Rotate overlay</li><li>RR: Distort overlay</li><li>T: Transparent overlay&nbsp;&nbsp;&nbsp;&nbsp;</li><li>DEL: Remove overlay</li></ul></b>"
+};
+```
+and sending it to the function below by defining a `customToolbarArray: [keymapper]` field during the `DistortableImageOverlay` object instantiation.
+
+```js
+function(map, position, el, styleClass, DOMString) {
+  var custom_toolbar = L.control({ position: position });
+
+  custom_toolbar.onAdd = function() {
+    var el_wrapper = L.DomUtil.create(el, styleClass);
+    el_wrapper.innerHTML = DOMString;
+    return el_wrapper;
+  };
+  custom_toolbar.addTo(map);
+}
+```
+
+## Usage
+
+```js
+// basic Leaflet map setup
+map = new L.map('map').setView([51.505, -0.09], 13);

```

Comment:
lets not redefine the map after the first time at the beginning of the README unless were showing something new regarding it